### PR TITLE
#5270: add support for isntalling build-deps via apt module

### PR DESCRIPTION
**Pull request by nathanhruby**

### CODE REVIEW

**Comment by nathanhruby** [comment]
ping?  Any chance of this getting in for 1.5?

**Comment by mpdehaan** [comment]
This didn't make 1.5 obviously though it seems reasonable, though it does seem the parameter needs to be "install_build_deps=yes/no" (a little more verbose) at least.

We'd have to test.

**Comment by thbourlove** [comment]
:+1:
any news about this PR?

File: library/packaging/apt [hunk 1]
```diff
@@ -82,6 +82,12 @@ options:
     required: false
     default: "yes"
     choices: [ "yes", "safe", "full", "dist"]
+  build_dep:
+    description:
+      - Instead, install the build dependancies for the named pkg (equivalent to 'apt-get build-dep foo')
```
Comment (yarikoptic) [inline library/packaging/apt]: s/dependancies/dependencies/

**Comment by yarikoptic** [comment]
hm -- there is no unittests in ansible for commands? ... there are: `test/integration/roles/test_apt/tasks/apt.yml` -- so I guess those are due for this one?

**Comment by nathanhruby** [comment]
@yarikoptic I believe those tests were not there when I originally wrote this patch.  Will investigate.

**Comment by nathanhruby** [comment]
@mpdehaan "install_build_deps=yes/no" seems to go against the grain for the apt module in that it seems to closely mirror the verbiage and constructs of apt.  In this case the "build-dep" arg seems to work better.  Happy to change it if you say so, but I think longer version seems out of place.

**Comment by nathanhruby** [comment]
Addressed typo, rebased to current devel to address merge conflicts.

**Comment by nathanhruby** [comment]
added tests, fixed another bug.  remaining item is name issue.

**Comment by jensenbox** [comment]
This is such a needed function for when building virtualenv modules in python.

Can't wait to see it make the light of day!

**Comment by nathanhruby** [comment]
ping: any chance of this making 1.7?

**Comment by mpdehaan** [comment]
# Cleanup Request

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us. 

We do require submissions to be free of merge commits, which cause difficulties in reviewing and merging code, and also affect the history of the source tree.
To resolve this problem, it may be helpful to create a new branch and cherry-pick commits onto it.
- In general, always start a "topic branch" named after each thing you would like to contribute
- use "git pull --rebase" vs "git pull" to stay in tune with upstream, as well as "git rebase" vs "git merge"
- if you have things broken out over many commits, consider submitting just one squashed commit that keeps the history concise

Just as a quick reminder of things, this is a really busy project.  We have over 800 contributors and to manage the queue effectively
we assign things a priority between P1 (highest) and P5.  We'd like to thank you very much for your time!  
We'll work things in priority order, so just wanted you to be aware of the queue and know we haven't forgot about you!

We will definitely see your comments on this issue when reading this ticket, but may not be able to reply promptly.  You may also wish to join one of our two mailing lists
which is very active:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Thank you once again for this and your interest in Ansible!

**Comment by nathanhruby** [comment]
Doh, sorry.  I think I did that merge commit before reading the rebased-branch-only rule.  Will fix this up this shortly.

**Comment by nathanhruby** [comment]
rebased my branch and created a new PR #8583 with it, closing this PR in favor of that.

**Pull request closed without merging**